app/pages/admin.js

```diff
@@ -1,25 +1,49 @@
-import Head from "next/head";
-import Image from "next/image";
-import styles from "../styles/Home.module.css";
 import React, { useState, useEffect } from "react";
 import Link from "next/link";
-import { stringify } from "gray-matter";
 
 export default function Admin() {
-  const [links, setLinks] = useState([{id: 0, name: "", link: ""}]);
-  const [amountOfLinks, setAmount] = useState(0);
- 
-  const clickAddLink = () => {
-    setAmount(amountOfLinks + 1); 
-    setLinks(links => [...links, {id: 0, name: "", link: ""}]);
-    }
- 
-    function change (index)
-    {
-      links[index].name = document.getElementById("name");
-    }
-  const [Prev, setPrev] = useState(false);
-  const handleClickPrev = () => setPrev(!Prev);
+  const [link, setLink] = useState({ name: "", link: "" });
+  const [links, setLinks] = useState(new Map());
+
+  const clickAddLink = () => addToMap(links.size + 1, link);
+  const deleteOnMap = (key) => {
+    setLinks((prev) => {
+      const newState = new Map(prev);
+      newState.delete(key);
+      return newState;
+    });
+  };
+  
+ const resetLink = () =>
+ {
+  setLink({ ...link, name: '' })
+  setLink({ ...link, link: '' })
```

Comment:
Why call setter twice? You only have an object with two fields, you can reset with one call like so:
```suggestion
  setLink({ name: '', link: '' })
```